store/watcher_hub.go

```diff
@@ -61,7 +63,9 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index, storeInde
 	}
 
 	w := &watcher{
-		eventChan:  make(chan *Event, 100), // use a buffered channel
+		// The eventChan needs to be at least as big as the event history to support streaming
+		// events from an outdated index.
+		eventChan:  make(chan *Event, wh.EventHistory.Queue.Capacity),
```

Comment:
the capability of the chan is the length of the history size. it will per-allocate the memory of the pointers, right?